cinema/views.py

```diff
@@ -56,3 +93,54 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        date_str = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+        filters = {}
+        if date_str:
+            date = dt.strptime(date_str, "%Y-%m-%d")
+            filters["show_time__date"] = date
+        if movie:
+            filters["movie__id"] = int(movie)
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("cinema_hall")
+                .annotate(
+                    tickets_available=F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            ).order_by("id")
+        return queryset.filter(**filters)
```

Comment:
Ensure that the 'filters' dictionary is correctly applied to the queryset. The current implementation might not filter correctly if both 'date_str' and 'movie' are provided.